File: Algorithms/Sorting/BubbleSort/C/bubbleSort.c
#include <stdio.h>

void swap(int *a, int *b){
	int t=(*a);
	(*a)=(*b);
	(*b)=t;
}

void bublleSort(int *arr, int siz){
	int i, j;
	for(i=0; i<siz; i++){
		for (j = 0; j < siz-i; ++j)
		{
			if(arr[j]>arr[j+1]) swap(&arr[j], &arr[j+1]);
		}
	}
}

int main()
{
	int arr[] = {5,22,6,3,74,7,3,8,2334,6}, i;
	bublleSort(arr, 10);
	for(i=0; i<10; i++) printf("%d\t", arr[i]);
	return 0;
}

File: Sorting/BubbleSort/bubble_sort.java
package dsa;

import java.util.Arrays;

public class bubble_sort {
    public static void main(String[] args) {
        int[] arr={-12,12,45,-67,-1,-99,-94,0,-3,99};
        sort(arr);
        System.out.println(Arrays.toString(arr));
    }
    static void sort(int[] arr){
//        boolean swap=false;
        int i=1;
        int a=arr.length;
        while(i<a){
            boolean swap=false;
            for(int j=0;j<a-i;j++){
                if(arr[j]>arr[j+1]){
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                    swap=true;
                }
            }
            if(!swap){
                break;
            }
            i=i+1;
        }
    }
}
